students/Ruohan/lesson04/mailroom_part2.py

```diff
@@ -0,0 +1,115 @@
+#! /usr/bin/env python3
+
+import sys
+import string
+
+old_datas = [['William Gates III',  653784.39, 2],
+         ['Mark Zuckerberg', 16396.10, 3],
+         ['Jeff Bezos', 877.33, 1],
+         ['Paul Allen', 708.42, 3]]
+
+# make a new data structure for part2
+datas = []
+
+for i in old_datas:
+    dic = {}
+    dic.setdefault('name', i[0])
```

Comment:
do you need setdefault here? why not just set it:

dic['name] = i[0]

you could use some nifty tricks here:

`dic = dict(zip(('name', 'donation_amount', 'donation_times'), i))`

(untested)